internal/schema/schema.go

```diff
@@ -423,6 +442,39 @@ func (s *Schema) GetQueryStringForMutation(mutation *Field, depth int) string {
 	return final
 }
 
+// RecursiveLookupFieldByPath traverses the GraphQL query types
+// based on the provided query fields path which represents a
+// GraphQL query. This method returns the last field of the last "node"
+// of the query tree "branch" - i.e. actor.apiAccess.key where `key`
+// is the primary target "leaf".
+//
+// e.g. `query { actor { apiAccess { key }}}` would have a path of ["actor", "apiAccess", "key"]
+func (s *Schema) RecursiveLookupFieldByPath(queryFieldPath []string, obj *Type) *Field {
```

Comment:
I think if you are after the Type for the last element in the query path, this might work: https://github.com/newrelic/tutone/blob/master/internal/schema/schema.go#L155

We can always merge and adjust later, just pointing it out in case it's useful.

Reply:
I actually used that at the start, but it gave me _all_ of the types, which I would then need to select from. I only needed the last leaf so I made a method that returns a single `Field` struct versus a potentially large slice.

Reference to my prior usage: https://github.com/newrelic/tutone/blob/bdb1f5632f5ed17c19a1768328a29136f819058f/generators/command/command_util.go#L65-L68